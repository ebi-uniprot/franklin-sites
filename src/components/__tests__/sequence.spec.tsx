import { screen, render, fireEvent } from '@testing-library/react';

import Sequence from '../sequence';

import sequenceData from '../../mock-data/sequence-data';

describe('Sequence component', () => {
  let rendered: ReturnType<typeof render>;

  beforeEach(() => {
    rendered = render(
      <Sequence
        sequence={sequenceData}
        accession="P05067"
        downloadUrl="https://url-to-P05067.fasta"
      />
    );
  });

  test('should render', () => {
    const { asFragment } = rendered;
    expect(asFragment()).toMatchSnapshot();
  });

  test('should render without tools', () => {
    const { asFragment } = render(
      <Sequence sequence={sequenceData} showActionBar={false} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Sequence component show/hide', () => {
  const handleSequenceLoad = jest.fn();

  test('should toggle view/hide on loaded sequence', () => {
    render(
      <Sequence sequence={sequenceData} accession="P05067" isCollapsible />
    );
    // Look for 'MFNFPHPAID' as it's the first chunk of 10 in the mock sequence
    expect(screen.queryByText('MFNFPHPAID')).not.toBeInTheDocument();
    fireEvent.click(screen.getByRole('button', { name: /Show sequence/ }));
    expect(screen.getByText('MFNFPHPAID')).toBeInTheDocument();
    fireEvent.click(screen.getByRole('button', { name: /Hide sequence/ }));
    expect(screen.queryByText('MFNFPHPAID')).not.toBeInTheDocument();
  });

  test('should be collapsed and trigger loading of sequence', () => {
    render(<Sequence onShowSequence={handleSequenceLoad} accession="P05067" />);
    fireEvent.click(screen.getByRole('button', { name: /Show/ }));
    expect(handleSequenceLoad).toBeCalled();
  });
});
