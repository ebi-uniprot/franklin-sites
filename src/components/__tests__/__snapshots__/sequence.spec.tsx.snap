// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sequence component should render 1`] = `
<DocumentFragment>
  <div>
    <section
      class="sequence-container"
    >
      <div
        class="action-bar button-group"
      >
        <div
          aria-expanded="false"
          aria-haspopup="true"
          class="dropdown"
        >
          <button
            class="button tertiary"
            type="button"
          >
            Tools
          </button>
        </div>
        <a
          class="button tertiary"
          download=""
          href="https://url-to-P05067.fasta"
        >
          <test-file-stub />
          Download
        </a>
        <div
          aria-expanded="false"
          aria-haspopup="true"
          class="dropdown"
        >
          <button
            class="button tertiary"
            type="button"
          >
            Highlight
          </button>
        </div>
      </div>
      <div
        class="sequence"
      >
        <span
          class="sequence__chunk"
        >
          MFNFPHPAID
        </span>
        <span
          class="sequence__chunk"
        >
          LASRMKSSPL
        </span>
        <span
          class="sequence__chunk"
        >
          MAGGSSSASS
        </span>
        <span
          class="sequence__chunk"
        >
          EDLFSPPMME
        </span>
        <span
          class="sequence__chunk"
        >
          DLDTPMTEYP
        </span>
        <span
          class="sequence__chunk"
        >
          MGSPPRMPYR
        </span>
        <span
          class="sequence__chunk"
        >
          GEDIEIAFLR
        </span>
        <span
          class="sequence__chunk"
        >
          SEASIKKSSL
        </span>
        <span
          class="sequence__chunk"
        >
          FNDKFAATLD
        </span>
        <span
          class="sequence__chunk"
        >
          DLSARPIDSA
        </span>
        <span
          class="sequence__chunk"
        >
          SLIGKLQSMT
        </span>
        <span
          class="sequence__chunk"
        >
          RSVREILDSG
        </span>
        <span
          class="sequence__chunk"
        >
          GEDIEIAFLR
        </span>
        <span
          class="sequence__chunk"
        >
          SEASIKKSSL
        </span>
        <span
          class="sequence__chunk"
        >
          FNDKFAATLD
        </span>
        <span
          class="sequence__chunk"
        >
          DLSARPIDSA
        </span>
        <span
          class="sequence__chunk"
        >
          SLIGKLQSMT
        </span>
        <span
          class="sequence__chunk"
        >
          RSVREILDSG
        </span>
        <span
          class="sequence__chunk"
        >
          DQLVHEDGPQ
        </span>
        <span
          class="sequence__chunk"
        >
          EILKQFVRVV
        </span>
        <span
          class="sequence__chunk"
        >
          NKHLCQDEDI
        </span>
        <span
          class="sequence__chunk"
        >
          HTVLAPLALE
        </span>
        <span
          class="sequence__chunk"
        >
          PEEKFHIIQT
        </span>
        <span
          class="sequence__chunk"
        >
          YYQAISMTQF
        </span>
        <span
          class="sequence__chunk"
        >
          DQLVHEDGPQ
        </span>
        <span
          class="sequence__chunk"
        >
          EILKQFVRVV
        </span>
        <span
          class="sequence__chunk"
        >
          NKHLCQDEDI
        </span>
        <span
          class="sequence__chunk"
        >
          HTVLAPLALE
        </span>
        <span
          class="sequence__chunk"
        >
          PEEKFHIIQT
        </span>
        <span
          class="sequence__chunk"
        >
          YYQAISMTQF
        </span>
        <span
          class="sequence__chunk"
        >
          VSPKWTSSLL
        </span>
        <span
          class="sequence__chunk"
        >
          SDALCRRANI
        </span>
        <span
          class="sequence__chunk"
        >
          VTVFNGQGVE
        </span>
        <span
          class="sequence__chunk"
        >
          GYFSELQHLY
        </span>
        <span
          class="sequence__chunk"
        >
          DTYGGLLAEP
        </span>
        <span
          class="sequence__chunk"
        >
          LYALSKQLKG
        </span>
        <span
          class="sequence__chunk"
        >
          VSPKWTSSLL
        </span>
        <span
          class="sequence__chunk"
        >
          SDALCRRANI
        </span>
        <span
          class="sequence__chunk"
        >
          VTVFNGQGVE
        </span>
        <span
          class="sequence__chunk"
        >
          GYFSELQHLY
        </span>
        <span
          class="sequence__chunk"
        >
          DTYGGLLAEP
        </span>
        <span
          class="sequence__chunk"
        >
          LYALSKQLKG
        </span>
        <span
          class="sequence__chunk"
        >
          LASDVRAQDM
        </span>
        <span
          class="sequence__chunk"
        >
          YPHGLDVIGW
        </span>
        <span
          class="sequence__chunk"
        >
          LENPEARPST
        </span>
        <span
          class="sequence__chunk"
        >
          DYLLSAPVSQ
        </span>
        <span
          class="sequence__chunk"
        >
          PLIGLVQLLN
        </span>
        <span
          class="sequence__chunk"
        >
          YAITCKILNK
        </span>
        <span
          class="sequence__chunk"
        >
          LASDVRAQDM
        </span>
        <span
          class="sequence__chunk"
        >
          YPHGLDVIGW
        </span>
        <span
          class="sequence__chunk"
        >
          LENPEARPST
        </span>
        <span
          class="sequence__chunk"
        >
          DYLLSAPVSQ
        </span>
        <span
          class="sequence__chunk"
        >
          PLIGLVQLLN
        </span>
        <span
          class="sequence__chunk"
        >
          YAITCKILNK
        </span>
        <span
          class="sequence__chunk"
        >
          SPGEFARHLS
        </span>
        <span
          class="sequence__chunk"
        >
          GSAGHSQGIV
        </span>
        <span
          class="sequence__chunk"
        >
          VAAMLATVVS
        </span>
        <span
          class="sequence__chunk"
        >
          WPTFFDAAST
        </span>
        <span
          class="sequence__chunk"
        >
          ALQVLFWIGC
        </span>
        <span
          class="sequence__chunk"
        >
          RSQQCYPSHS
        </span>
        <span
          class="sequence__chunk"
        >
          SPGEFARHLS
        </span>
        <span
          class="sequence__chunk"
        >
          GSAGHSQGIV
        </span>
        <span
          class="sequence__chunk"
        >
          VAAMLATVVS
        </span>
        <span
          class="sequence__chunk"
        >
          WPTFFDAAST
        </span>
        <span
          class="sequence__chunk"
        >
          ALQVLFWIGC
        </span>
        <span
          class="sequence__chunk"
        >
          RSQQCYPSHS
        </span>
        <span
          class="sequence__chunk"
        >
          IPPSLVDQSE
        </span>
        <span
          class="sequence__chunk"
        >
          RLSPMLSVKG
        </span>
        <span
          class="sequence__chunk"
        >
          ASRESLLKYL
        </span>
        <span
          class="sequence__chunk"
        >
          DEHNRHLPPA
        </span>
        <span
          class="sequence__chunk"
        >
          QQGSLALING
        </span>
        <span
          class="sequence__chunk"
        >
          RQQFVVAGNP
        </span>
        <span
          class="sequence__chunk"
        >
          IPPSLVDQSE
        </span>
        <span
          class="sequence__chunk"
        >
          RLSPMLSVKG
        </span>
        <span
          class="sequence__chunk"
        >
          ASRESLLKYL
        </span>
        <span
          class="sequence__chunk"
        >
          DEHNRHLPPA
        </span>
        <span
          class="sequence__chunk"
        >
          QQGSLALING
        </span>
        <span
          class="sequence__chunk"
        >
          RQQFVVAGNP
        </span>
        <span
          class="sequence__chunk"
        >
          LSLYAFANKL
        </span>
        <span
          class="sequence__chunk"
        >
          RAASNNSSTT
        </span>
        <span
          class="sequence__chunk"
        >
          NTARVPFSQR
        </span>
        <span
          class="sequence__chunk"
        >
          PLLITARFLP
        </span>
        <span
          class="sequence__chunk"
        >
          ISVPFHTSLL
        </span>
        <span
          class="sequence__chunk"
        >
          EDAEAQILED
        </span>
        <span
          class="sequence__chunk"
        >
          ED
        </span>
      </div>
    </section>
  </div>
</DocumentFragment>
`;

exports[`Sequence component should render without tools 1`] = `
<DocumentFragment>
  <div>
    <section
      class="sequence-container"
    >
      <div
        class="sequence"
      >
        <span
          class="sequence__chunk"
        >
          MFNFPHPAID
        </span>
        <span
          class="sequence__chunk"
        >
          LASRMKSSPL
        </span>
        <span
          class="sequence__chunk"
        >
          MAGGSSSASS
        </span>
        <span
          class="sequence__chunk"
        >
          EDLFSPPMME
        </span>
        <span
          class="sequence__chunk"
        >
          DLDTPMTEYP
        </span>
        <span
          class="sequence__chunk"
        >
          MGSPPRMPYR
        </span>
        <span
          class="sequence__chunk"
        >
          GEDIEIAFLR
        </span>
        <span
          class="sequence__chunk"
        >
          SEASIKKSSL
        </span>
        <span
          class="sequence__chunk"
        >
          FNDKFAATLD
        </span>
        <span
          class="sequence__chunk"
        >
          DLSARPIDSA
        </span>
        <span
          class="sequence__chunk"
        >
          SLIGKLQSMT
        </span>
        <span
          class="sequence__chunk"
        >
          RSVREILDSG
        </span>
        <span
          class="sequence__chunk"
        >
          GEDIEIAFLR
        </span>
        <span
          class="sequence__chunk"
        >
          SEASIKKSSL
        </span>
        <span
          class="sequence__chunk"
        >
          FNDKFAATLD
        </span>
        <span
          class="sequence__chunk"
        >
          DLSARPIDSA
        </span>
        <span
          class="sequence__chunk"
        >
          SLIGKLQSMT
        </span>
        <span
          class="sequence__chunk"
        >
          RSVREILDSG
        </span>
        <span
          class="sequence__chunk"
        >
          DQLVHEDGPQ
        </span>
        <span
          class="sequence__chunk"
        >
          EILKQFVRVV
        </span>
        <span
          class="sequence__chunk"
        >
          NKHLCQDEDI
        </span>
        <span
          class="sequence__chunk"
        >
          HTVLAPLALE
        </span>
        <span
          class="sequence__chunk"
        >
          PEEKFHIIQT
        </span>
        <span
          class="sequence__chunk"
        >
          YYQAISMTQF
        </span>
        <span
          class="sequence__chunk"
        >
          DQLVHEDGPQ
        </span>
        <span
          class="sequence__chunk"
        >
          EILKQFVRVV
        </span>
        <span
          class="sequence__chunk"
        >
          NKHLCQDEDI
        </span>
        <span
          class="sequence__chunk"
        >
          HTVLAPLALE
        </span>
        <span
          class="sequence__chunk"
        >
          PEEKFHIIQT
        </span>
        <span
          class="sequence__chunk"
        >
          YYQAISMTQF
        </span>
        <span
          class="sequence__chunk"
        >
          VSPKWTSSLL
        </span>
        <span
          class="sequence__chunk"
        >
          SDALCRRANI
        </span>
        <span
          class="sequence__chunk"
        >
          VTVFNGQGVE
        </span>
        <span
          class="sequence__chunk"
        >
          GYFSELQHLY
        </span>
        <span
          class="sequence__chunk"
        >
          DTYGGLLAEP
        </span>
        <span
          class="sequence__chunk"
        >
          LYALSKQLKG
        </span>
        <span
          class="sequence__chunk"
        >
          VSPKWTSSLL
        </span>
        <span
          class="sequence__chunk"
        >
          SDALCRRANI
        </span>
        <span
          class="sequence__chunk"
        >
          VTVFNGQGVE
        </span>
        <span
          class="sequence__chunk"
        >
          GYFSELQHLY
        </span>
        <span
          class="sequence__chunk"
        >
          DTYGGLLAEP
        </span>
        <span
          class="sequence__chunk"
        >
          LYALSKQLKG
        </span>
        <span
          class="sequence__chunk"
        >
          LASDVRAQDM
        </span>
        <span
          class="sequence__chunk"
        >
          YPHGLDVIGW
        </span>
        <span
          class="sequence__chunk"
        >
          LENPEARPST
        </span>
        <span
          class="sequence__chunk"
        >
          DYLLSAPVSQ
        </span>
        <span
          class="sequence__chunk"
        >
          PLIGLVQLLN
        </span>
        <span
          class="sequence__chunk"
        >
          YAITCKILNK
        </span>
        <span
          class="sequence__chunk"
        >
          LASDVRAQDM
        </span>
        <span
          class="sequence__chunk"
        >
          YPHGLDVIGW
        </span>
        <span
          class="sequence__chunk"
        >
          LENPEARPST
        </span>
        <span
          class="sequence__chunk"
        >
          DYLLSAPVSQ
        </span>
        <span
          class="sequence__chunk"
        >
          PLIGLVQLLN
        </span>
        <span
          class="sequence__chunk"
        >
          YAITCKILNK
        </span>
        <span
          class="sequence__chunk"
        >
          SPGEFARHLS
        </span>
        <span
          class="sequence__chunk"
        >
          GSAGHSQGIV
        </span>
        <span
          class="sequence__chunk"
        >
          VAAMLATVVS
        </span>
        <span
          class="sequence__chunk"
        >
          WPTFFDAAST
        </span>
        <span
          class="sequence__chunk"
        >
          ALQVLFWIGC
        </span>
        <span
          class="sequence__chunk"
        >
          RSQQCYPSHS
        </span>
        <span
          class="sequence__chunk"
        >
          SPGEFARHLS
        </span>
        <span
          class="sequence__chunk"
        >
          GSAGHSQGIV
        </span>
        <span
          class="sequence__chunk"
        >
          VAAMLATVVS
        </span>
        <span
          class="sequence__chunk"
        >
          WPTFFDAAST
        </span>
        <span
          class="sequence__chunk"
        >
          ALQVLFWIGC
        </span>
        <span
          class="sequence__chunk"
        >
          RSQQCYPSHS
        </span>
        <span
          class="sequence__chunk"
        >
          IPPSLVDQSE
        </span>
        <span
          class="sequence__chunk"
        >
          RLSPMLSVKG
        </span>
        <span
          class="sequence__chunk"
        >
          ASRESLLKYL
        </span>
        <span
          class="sequence__chunk"
        >
          DEHNRHLPPA
        </span>
        <span
          class="sequence__chunk"
        >
          QQGSLALING
        </span>
        <span
          class="sequence__chunk"
        >
          RQQFVVAGNP
        </span>
        <span
          class="sequence__chunk"
        >
          IPPSLVDQSE
        </span>
        <span
          class="sequence__chunk"
        >
          RLSPMLSVKG
        </span>
        <span
          class="sequence__chunk"
        >
          ASRESLLKYL
        </span>
        <span
          class="sequence__chunk"
        >
          DEHNRHLPPA
        </span>
        <span
          class="sequence__chunk"
        >
          QQGSLALING
        </span>
        <span
          class="sequence__chunk"
        >
          RQQFVVAGNP
        </span>
        <span
          class="sequence__chunk"
        >
          LSLYAFANKL
        </span>
        <span
          class="sequence__chunk"
        >
          RAASNNSSTT
        </span>
        <span
          class="sequence__chunk"
        >
          NTARVPFSQR
        </span>
        <span
          class="sequence__chunk"
        >
          PLLITARFLP
        </span>
        <span
          class="sequence__chunk"
        >
          ISVPFHTSLL
        </span>
        <span
          class="sequence__chunk"
        >
          EDAEAQILED
        </span>
        <span
          class="sequence__chunk"
        >
          ED
        </span>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
